src/server/twemcache-rs/src/protocol/data/parser.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! Memcache data protocol parsing
+
+use bytes::BytesMut;
+use std::borrow::Borrow;
+
+use super::*;
+
+// TODO(bmartin): this should be lifted out into a common crate and shared
+// between different protocols
+pub trait Parser {
```

Comment:
Another TODO would be to add a parser to responses (you probably have it in rpc-perf already?) to complete the loop